plugin/tmux_navigator.vim

```diff
@@ -67,12 +67,16 @@ function! s:TmuxAwareNavigate(direction)
   " a) we're toggling between the last tmux pane;
   " b) we tried switching windows in vim but it didn't have effect.
   if tmux_last_pane || nr == winnr()
-    if g:tmux_navigator_save_on_switch
-      try
-        update
-      catch /^Vim\%((\a\+)\)\=:E32/
-      endtry
-    endif
+    try
+      if g:tmux_navigator_save_on_switch == 1
+          update " save the active buffer. See :help update
+      elseif g:tmux_navigator_save_on_switch == 2
+          wall "  save all the buffers. See :help wall
+      endif
+    catch /^Vim\%((\a\+)\)\=:E32/ " catches the no file name error thrown by update
+    catch /^Vim\%((\a\+)\)\=:E141/ " catches the no file name error thrown by wall
+    endtry
```

Comment:
You might want to put the command in their own `try` blocks (and only catch the relevant error).